src/SWP/Bundle/TemplateEngineBundle/Controller/MenuController.php

```diff
@@ -0,0 +1,235 @@
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\Route;
+use SWP\Bundle\TemplateEngineBundle\Provider\TenantAwareMenuProvider;
+use Symfony\Cmf\Bundle\MenuBundle\Doctrine\Phpcr\Menu;
+use Symfony\Component\HttpFoundation\Request;
+use SWP\Bundle\TemplateEngineBundle\Form\Type\MenuType;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+use Symfony\Component\HttpKernel\Exception\UnprocessableEntityHttpException;
+
+class MenuController extends FOSRestController
+{
+    /**
+     * Lists all registered menus.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Lists all registered menus",
+     *     statusCodes={
+     *         200="Returned on success."
+     *     }
+     * )
+     * @Route("/api/{version}/templates/menus/", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_list_menus")
+     * @Method("GET")
+     */
+    public function listAction(Request $request)
+    {
+        $menuParent = $this->getMenuParent();
+        $dm = $this->get('document_manager');
+        $menus = $dm->getChildren($menuParent);
+        $paginator = $this->get('knp_paginator');
+        $menus = $paginator->paginate($menus);
+
+        if (count($menus) == 0) {
+            throw new NotFoundHttpException('Menus were not found.');
+        }
+
+        return $this->handleView(View::create($this->container->get('swp_pagination_rep')->createRepresentation($menus, $request), 200));
+    }
+
+    /**
+     * Get single menu.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Get single menu",
+     *     statusCodes={
+     *         200="Returned on success.",
+     *         404="Menu not found",
+     *         422="Menu id is not number"
+     *     }
+     * )
+     * @Route("/api/{version}/templates/menus/{id}", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_get_menu")
+     * @Method("GET")
+     */
+    public function getAction(Request $request, $id)
+    {
+        $menu = $this->getMenu($id);
+        if (!$menu) {
+            throw new NotFoundHttpException('Menu with this id was not found.');
+        }
+
+        return $this->handleView(View::create($menu, 200));
+    }
+
+    /**
+     * Create new menu.
+     *
+     * @ApiDoc(
+     *     resource=true,
+     *     description="Create new menu",
+     *     statusCodes={
+     *         201="Returned on success.",
+     *         400="Returned when form have errors"
+     *     },
+     *     input="SWP\Bundle\TemplateEngineBundle\Form\Type\MenuType"
+     * )
+     * @Route("/api/{version}/templates/menus", options={"expose"=true}, defaults={"version"="v1"}, name="swp_api_templates_create_menu")
+     * @Method("POST")
+     */
+    public function createAction(Request $request)
+    {
+        $menu = new Menu();
+
+        // Parent must be set before validation
+        $menuParent = $this->getMenuParent();
+        $menu->setParentDocument($menuParent);
```

Comment:
It seems like `$menuParent` defined by `$this->getMenuParent()` on line 104 can also be of type null; however, `Symfony\Cmf\Bundle\MenuB...nu::setParentDocument()` does only seem to accept object, maybe add an additional type check?